scripts/water-model/water-script.py

```diff
@@ -0,0 +1,203 @@
+
+# check = potable_assets.groupby(['node_id','WSZONEID'])['asset_connected_pop_2010'].count().to_frame().reset_index()
+# print(check[check['asset_connected_pop_2010']>1])
+# print(oliv)
+
+potable_facilities_NWC = gpd.read_file('/soge-home/projects/mistral/jamaica-ccri/processed_data/networks/water/potable_facilities_NWC.gpkg')
+potable_facilities_NWC = potable_facilities_NWC.drop(['asset_pop_new_2010','asset_pop_new_2020','asset_pop_new_2030'], axis=1)
+potable_facilities_NWC_pop = potable_facilities_NWC.merge(potable_assets[['node_id','WSZONEID','asset_connected_pop_2010']], on='node_id', how='left').drop_duplicates()
+#potable_facilities_NWC_pop['asset_connected_pop_2010'] = potable_facilities_NWC_pop['asset_connected_pop_2010'].apply(roundup)
+potable_facilities_NWC_pop.to_file('potable_facilities_NWC_pop.gpkg', driver='GPKG')
+
+pipelines_centroid = pd.read_csv('potable/processed/pipelines_centroid.csv')
+pipelines_centroid['OBJECTID'] = pd.to_numeric(pipelines_centroid['OBJECTID'])
+pipelines_centroid['edge_id'] = 'potable_'+pipelines_centroid['OBJECTID'].astype(float).astype(str)
+for it,row in pipelines_centroid.iterrows(): #canal/pipe centroid
+    pipeline_id = row.OBJECTID #OBJECTID
+    pipelines_Y = row.Y
+    pipelines_X = row.X
+    potable_assets['lat_diff'] = abs(potable_assets['Y'] - pipelines_Y)
+    potable_assets['lon_diff'] = abs(potable_assets['X'] - pipelines_X)
+    potable_assets['sqrt'] = np.sqrt(np.square(potable_assets['lat_diff'])+np.square(potable_assets['lat_diff']))
+    closest_wrz = potable_assets[potable_assets['sqrt']==potable_assets['sqrt'].min()]['WSZONEID'].values[0]
+    asset_connected_pop_2010 = potable_assets[potable_assets['sqrt']==potable_assets['sqrt'].min()]['asset_connected_pop_2010'].values[0]
+    asset_connected_pop_2030 = potable_assets[potable_assets['sqrt']==potable_assets['sqrt'].min()]['asset_connected_pop_2030'].values[0]
+    pipelines_centroid.loc[it,'WSZONEID'] = closest_wrz
+    pipelines_centroid.loc[it,'asset_connected_pop_2010'] = asset_connected_pop_2010
+    pipelines_centroid.loc[it,'asset_connected_pop_2030'] = asset_connected_pop_2030
+
+pipelines_NWC = gpd.read_file('/soge-home/projects/mistral/jamaica-ccri/processed_data/networks/water/pipelines_NWC.gpkg')
+pipelines_NWC_pop = pipelines_NWC.merge(pipelines_centroid[['edge_id','WSZONEID','asset_connected_pop_2010','asset_connected_pop_2030']], on='edge_id', how='left').drop_duplicates()
+#pipelines_NWC_pop = pipelines_NWC_pop.dropna(subset=['asset_connected_pop_2010'])
+#pipelines_NWC_pop['asset_connected_pop_2010'] = pipelines_NWC_pop['asset_connected_pop_2010'].apply(roundup)
+pipelines_NWC_pop.to_file('pipelines_NWC_edges_pop.gpkg', driver='GPKG')
+
+#########################################################################
+#########################################################################
+# disruption analysis potable
```

Comment:
Pull this and the irrigation disruption analysis into a separate script, which can take the damaged nodes as input.